ts/src/main/scala/ch/epfl/ts/brokers/ExampleBroker.scala

```diff
@@ -0,0 +1,129 @@
+package ch.epfl.ts.brokers
+
+import akka.actor.{ActorLogging, Props, ActorRef}
+import akka.pattern.ask
+import akka.util.Timeout
+import ch.epfl.ts.component.Component
+import ch.epfl.ts.data._
+import ch.epfl.ts.engine._
+import ch.epfl.ts.engine.GetWalletFunds
+import ch.epfl.ts.engine.WalletConfirm
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import ch.epfl.ts.data.{Order, MarketBidOrder}
+import ch.epfl.ts.engine.WalletConfirm
+import ch.epfl.ts.data.Register
+import ch.epfl.ts.engine.FundWallet
+import ch.epfl.ts.engine.WalletFunds
+import ch.epfl.ts.engine.GetWalletFunds
+import ch.epfl.ts.engine.WalletInsufficient
+import ch.epfl.ts.data.ConfirmRegistration
+import ch.epfl.ts.data.Currency._
+import ch.epfl.ts.engine.WalletConfirm
+import ch.epfl.ts.data.Register
+import ch.epfl.ts.engine.FundWallet
+import scala.Some
+import ch.epfl.ts.engine.WalletFunds
+import ch.epfl.ts.engine.GetWalletFunds
+import ch.epfl.ts.engine.WalletInsufficient
+import ch.epfl.ts.data.Currency
+import ch.epfl.ts.data.ConfirmRegistration
+
+/**
+ * Created by sygi on 03.04.15.
+ */
+class ExampleBroker extends Component with ActorLogging { //TODO(sygi): println -> log.debug
+  import context.dispatcher
+  var mapping = Map[Long, ActorRef]()
+  override def receiver: PartialFunction[Any, Unit] = {
+    case Register(id) => {
+      log.debug("Broker: registration of agent " + id)
+      log.debug("with ref: " + sender())
+      if (mapping.get(id) != None){
+        println("Duplicate Id")
+        //TODO(sygi): reply to the trader that registration failed
+        //TODO(sygi): return?
+      }
+      mapping = mapping + (id -> sender())
+      context.actorOf(Props[Wallet], "wallet" + id)
+      sender() ! ConfirmRegistration
+    }
+    case FundWallet(uid, curr, value) => {
+      log.debug("Broker: got a request to fund a wallet")
+      val replyTo = sender
+      executeForWallet(uid, FundWallet(uid, curr, value), {
+        case WalletConfirm(uid) => {
+          println("Broker: Wallet confirmed")
+          replyTo ! WalletConfirm(uid)
+        }
+        case WalletInsufficient(uid) => {
+          println("Broker: insufficient funds")
+          replyTo ! WalletInsufficient(uid)
+        }
+        case p => println("FundWallet: Unexpected message " + p)
+      })
+    }
+    case GetWalletFunds(uid) => { //TODO(sygi): check if trader asks for his wallet
+      log.debug("Broker: got a get show wallet request")
+      val replyTo = sender
+      executeForWallet(uid, GetWalletFunds(uid), {
+        case w: WalletFunds => {
+          replyTo ! w
+        }
+      })
+    }
+
+    //TODO(sygi): refactor charging the wallet/placing an order
+    //TODO(sygi): other orders
```

Comment:
Here you can use the `Chargeable` abstraction to support all order types at once, correct?